crates/katana/rpc/rpc/tests/saya.rs

```diff
@@ -202,3 +203,140 @@ async fn executions_chunks_logic_ok() {
 
     sequencer.stop().expect("failed to stop sequencer");
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fetch_traces_from_block() {
+    let sequencer = TestSequencer::start(
+        SequencerConfig { no_mining: true, ..Default::default() },
+        get_default_test_starknet_config(),
+    )
+    .await;
+
+    let client = HttpClientBuilder::default().build(sequencer.url()).unwrap();
+
+    let account = sequencer.account();
+
+    let path: PathBuf = PathBuf::from("tests/test_data/cairo1_contract.json");
+    let (contract, compiled_class_hash) =
+        common::prepare_contract_declaration_params(&path).unwrap();
+    let contract = Arc::new(contract);
+
+    let declare_res = account.declare(contract.clone(), compiled_class_hash).send().await.unwrap();
+
+    let max_fee = FieldElement::from_hex_be(ENOUGH_GAS).unwrap();
+    let mut nonce = FieldElement::ONE;
+
+    // Store the tx hashes to check the retrieved traces later.
+    let mut tx_hashes = vec![declare_res.transaction_hash];
+
+    // Prepare 29 transactions to test chunks (30 at total with the previous declare).
+    for i in 0..29 {
+        let call =
+            common::build_deploy_cairo1_contract_call(declare_res.class_hash, (i + 2_u32).into());
+
+        // we set the nonce manually so that we can send the tx rapidly without waiting for the
+        // previous tx to be mined first.
+        let res = account
+            .execute(vec![call])
+            .nonce(nonce)
+            .max_fee(max_fee)
+            .send()
+            .await
+            .expect("failed to send tx");
+
+        nonce += FieldElement::ONE;
+        tx_hashes.push(res.transaction_hash);
+    }
+
+    // Wait for the for all the txs to be mined first.
+    tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
Isn't better to use a transaction waiter on the latest transaction (as txs are sequentially inserted into the queue) instead of waiting an arbitrary time that may delay the tests?

Reply:
true, thought about it as well but rly i just copied from the existing test in the file lol. will change it